src/main/java/org/isf/patvac/rest/PatVacController.java

```diff
@@ -0,0 +1,159 @@
+import org.isf.patvac.manager.PatVacManager;
+import org.isf.patvac.mapper.PatVacMapper;
+import org.isf.patvac.model.PatientVaccine;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.isf.vaccine.manager.VaccineBrowserManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/patientvaccines", produces = MediaType.APPLICATION_JSON_VALUE)
+public class PatVacController {
+
+	@Autowired
+	protected PatVacManager patVacManager;
+	
+	@Autowired
+	protected VaccineBrowserManager vaccineManager;
+	
+	@Autowired
+	protected PatVacMapper mapper;
+
+	private final Logger logger = LoggerFactory.getLogger(PatVacController.class);
+
+	public PatVacController(PatVacManager patVacManager, PatVacMapper patientVaccinemapper) {
+		this.patVacManager = patVacManager;
+		this.mapper = patientVaccinemapper;
+	}
+
+	/**
+	 * create a new {@link PatientVaccine}
+	 * @param patientVaccineDTO
+	 * @return <code>true</code> if the operation type has been stored, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/patientvaccines", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<Integer> newPatientVaccine(@RequestBody PatientVaccineDTO patientVaccineDTO) throws OHServiceException {
+		int code = patientVaccineDTO.getCode();
+		logger.info("Create patient vaccine " + code);
+		boolean isCreated = patVacManager.newPatientVaccine(mapper.map2Model(patientVaccineDTO));
+		if (!isCreated) {
+			throw new OHAPIException(new OHExceptionMessage(null, "patient vaccine is not created!", OHSeverityLevel.ERROR));
+		}
+		return ResponseEntity.status(HttpStatus.CREATED).body(code);
+	}
+
+	/**
+	 * Updates the specified {@link PatientVaccine}.
+	 * @param patientVaccineDTO
+	 * @return <code>true</code> if the operation type has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/patientvaccines", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<Integer> updatePatientVaccinet(@RequestBody PatientVaccineDTO patientVaccineDTO)
```

Comment:
Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/patientvaccines/{code}", produces = MediaType.APPLICATION_JSON_VALUE)`